README.md

```diff
@@ -100,6 +101,108 @@ $ mqtt pub -t 'smartylighting/streetlights/1/0/event/123/lighting/measured' -h '
 #Notice that the server automatically validates incoming messages and logs out validation errors
 ```
 
+### Adding custom code / handlers
+
+It's highly recommended to treat the generated template as a library or API for initializing the server and integrating user-written handlers. Instead of directly modifying the template, leveraging it in this manner ensures that its regenerative capability is preserved. Any modifications made directly to the template would be overwritten upon regeneration.
+
+Consider a scenario where you intend to introduce a new channel or section to the AsyncAPI file, followed by a template regeneration. In this case, any modifications applied within the generated code would be overwritten.
+
+To avoid this, user code remains external to the generated code, functioning as an independent entity that consumes the generated code as a library. By adopting this approach, the user code remains unaffected during template regenerations.
+
+Facilitating this separation involves creating handlers and associating them with their respective routes. These handlers can then be seamlessly integrated into the template's workflow by importing the appropriate methods to register the handlers. In doing so, the template's `client.<operationId>` method becomes the bridge between the user-written handlers and the generated code. This can be used to register middlewares for specific methods on specific channels.
+
+> The AsyncAPI file used for the example is [here](https://bit.ly/asyncapi)
+
+```js
+// output refers to the generated template folder
+// You require the generated server. Running this code starts the server
+// App exposes API to send messages
+const { client } = require("./output");
+
+// to start the app
+client.init();
+
+// Generated handlers that we use to react on consumer / produced messages are attached to the client
+// through which we can register middleware functions
+
+/**
+ * 
+ * 
+ * Example of how to process a message before it is sent to the broker
+ * 
+ * 
+ */
+function testPublish() {
+    // mosquitto_sub -h test.mosquitto.org -p 1883 -t "smartylighting/streetlights/1/0/action/12/turn/on"
+
+    // Registering your custom logic in a channel-specific handler
+    // the passed handler function is called once the app sends a message to the channel
+    // For example `client.app.send` sends a message to some channel using and before it is sent, you want to perform some other actions
+    // in such a case, you can register middlewares like below
+    client.turnOn((message) => { // `turnOn` is the respective operationId
```

Comment:
we cannot call it this way as in the end it is not a function to "send" but just to act as middleware

and it will conflict with https://github.com/asyncapi/nodejs-template/issues/226

instead of `turnOn`, we should have `registerTurnOnMiddleware` or just `turnOnMiddleware`

Reply:
Looked into #226 , Makes complete sense, let's proceed by calling it `register<operationId>Middleware`

Do you also want me to complete #226 before the integration test ?

Reply:
up to you, not a must here.

Reply:
@kaushik-rishi please specify what is your approach here. Cause from my side, this PR is ready and we can merge, unless you want to solve #226 here too

Reply:
This is ready to merge.